draft-ietf-tls-extended-key-update.md

```diff
@@ -993,151 +993,144 @@ The following variables and abbreviations are used in the state machine diagrams
-                          |  old_rx=rx;          |
-                          |  retain_old=1;       |
-                          |  rx=rx+1; tx=tx+1    |
-                          +----------------------+
-                                   |
-                        (8) send ACK [tag=tx]
-                        set updating=0; assert tx==rx
-                                   |
-                                   |
-                                   v
-                          +----------------+
- APP send/recv allowed -- |   FINISHED     |
- retain_old=0 afterwards  +----------------+
++---------------------------------+
+| START                           |
+| rx := E; tx := E, updating := 0 |
++---------------------------------+
+      |
+(1) send Req [tag=tx]
+    set updating := 1
+      v
++------------------------------+
+| WAIT_RESP                    |
++------------------------------+
+ ^    ^    ^           ^
+ |    |    |           |
+ |    |    |           |
+ |    |    |           +-- (A) recv peer Req (crossed):
+ |    |    |                if peer < local  -> IGNORE (Self-Loop)
+ |    |    |                if peer == local -> ABORT  (Error)
+ |    |    |                if peer > local  -> ABANDON update;
+ |    |    |                                    updating := 0;
+ |	  |	   |                                    act as RESPONDER
+ |    |    |
+ |    |    +-- APP send (anytime)           (Self-Loop)
+ |    |
+ |    +-- APP recv: retain_old && e == old_rx   (Self-Loop)
+ |
+ +-- APP recv: e == rx                           (Self-Loop)
+
+          | recv Resp [e == rx]
+          | derive secrets; send NKU [tag=old tx]
+          v
++------------------------------+
+| WAIT_R_NKU                   |
++------------------------------+
+      |
+      |  APP send/recv allowed
+      |
+(6) recv NKU [e==rx]
+(Responder still tags old tx)
+      v
++----------------------+
+| ACTIVATE RETENTION   |
+| old_rx=rx;           |
+| retain_old=1;        |
+| rx=rx+1; tx=tx+1     |
++----------------------+
+      |
+(8) send ACK [tag=tx]
+    set updating := 0; assert tx==rx
+      v
++------------------------------------------+
+| FINISHED                                 |
+| (retain_old=0 after first APP at new rx) |
++------------------------------------------+
 ~~~
 
 ### State Machine (Responder)
 
-The responder starts in the START state with synchronized transmit and receive epochs (rx=tx=E) and no update in progress. Application data can be sent at any time with the current transmit epoch and is accepted if the epoch matches the receiver's view or, if retention is active, the previous epoch.
+The responder starts in the START state with synchronized transmit and receive epochs (rx := E; tx := E) and no update in progress. Application data can be sent at any time with the current transmit epoch and is accepted if the epoch matches the receiver's view or, if retention is active, the previous epoch.
```

Comment:
I can't parse the last line. you may want to say:
Application data can be transmitted at any time using the sender’s current transmit epoch. A receiver must accept application data if the epoch tag on the DTLS record equals the receiver’s current receive epoch. If the receiver has retention active (retain_old == true), the receiver must also accept DTLS records whose epoch tag equals the remembered previous epoch.

Reply:
I updated the text based in your suggestion.